extensions/amp-story/1.0/amp-story-reaction.js

```diff
@@ -52,102 +52,83 @@ const ENDPOINT_INVALID_ERROR =
  *    selectedByUser: boolean,
  * }}
  */
-export let ReactionType;
+export let ReactionOptionType;
 
 /**
  * @typedef {{
  *    totalResponseCount: number,
  *    hasUserResponded: boolean,
- *    responses: !Array<ReactionType>,
+ *    responses: !Array<ReactionOptionType>,
  * }}
  */
 export let ReactionResponseType;
 
 /**
- * Generates the template for the quiz.
- *
- * @param {!Element} element
- * @return {!Element}
+ * Reaction abstract class with shared functionality.
+ * @abstract
  */
-const buildQuizTemplate = (element) => {
-  const html = htmlFor(element);
-  return html`
-    <div class="i-amphtml-story-quiz-container">
-      <div class="i-amphtml-story-quiz-prompt-container"></div>
-      <div class="i-amphtml-story-quiz-option-container"></div>
-    </div>
-  `;
-};
-
-/**
- * Generates the template for each option.
- *
- * @param {!Element} option
- * @return {!Element}
- */
-const buildOptionTemplate = (option) => {
-  const html = htmlFor(option);
-  return html`
-    <span class="i-amphtml-story-quiz-option">
-      <span class="i-amphtml-story-quiz-answer-choice"></span>
-    </span>
-  `;
-};
-
-export class AmpStoryQuiz extends AMP.BaseElement {
+export class AmpStoryReaction extends AMP.BaseElement {
```

Comment:
Not a comment about this line, but could you add some comments documentation to briefly describe the lifecycle of this component? Especially which classes are added and when. 

eg: 
1. Builds an empty template through method X
2. Retrieves the publisher configuration and appends the content
3. HTTP call to know if user already answered
4. Something happens, class X is added
5. ...

Followup question, do you think all the classes we add could be replaced by a state machine, exposed as an attribute? I think relying on `[state=answered]` might be easier to read than `.i-amphtml-story-reaction-answered`. I have no idea if that's possible with the current implementation though, and also, feel free to keep this for a followup PR. :))

Reply:
I'll add the lifecycle as a comment to the class, no problem.

As of the state attribute, it's not possible since many states are independent of each other, such as having data (from a backend) and being answered. While having data implies being responded, a reaction can be responded and not have data so each class has its purpose. We could use an attribute selected="false" or selected="true" for **each** option, but wouldn't advise to have it as a general component attribute (selected="2").

On a new PR I plan to revise the component, and can look more into this though.